File: output/GPT4o-Mini/xCodeEval_4000/Java/C++/4f4568b46200ccf0a5f252065ee65790.cpp
#include <iostream>
#include <fstream>
#include <cmath>
#include <vector>
#include <string>
#include <sstream>
#include <iomanip>
#include <cstdlib>
#include <stdexcept>
#include <gmpxx.h>

using namespace std;

class MyScanner {
    ifstream br;
    string line;
    stringstream st;

public:
    MyScanner(const string& file) {
        br.open(file);
        if (!br.is_open()) {
            cerr << "File not found" << endl;
            exit(42);
        }
    }

    MyScanner() {
        br.open("/dev/stdin");
    }

    string nextToken() {
        while (st.eof() || !st.good()) {
            getline(br, line);
            st.clear();
            st.str(line);
        }
        string token;
        st >> token;
        return token;
    }

    int nextInt() {
        return stoi(nextToken());
    }

    double nextDouble() {
        return stod(nextToken());
    }
};

class C {
    MyScanner in;
    ofstream out;

    const int shift = 100000;
    const int maxn = 300000000;

    const vector<int> prec = {

File: output/GPT4o-Mini/xCodeEval_4000/Java/Python/4f4568b46200ccf0a5f252065ee65790.py
import sys
from math import isqrt
from sympy import isprime

class MyScanner:
    def __init__(self, file=None):
        if file:
            self.br = open(file, 'r')
        else:
            self.br = sys.stdin

    def next_token(self):
        return self.br.readline().strip()

    def next_int(self):
        return int(self.next_token())

class C:
    shift = 100000
    maxn = 300000000

    prec = [0, 4784, 8978, 12981, 16901, 20732, 24523, 28249,
            31916, 35585, 39176, 42818, 46430, 49962, 53516, 57023, 60523,
            64040, 67489, 70985, 74417, 77879, 81285, 84679, 88055, 91433,
            94878, 98240, 101608, 104979, 108284, 111653, 115039, 118375,
